core/common/src/main/java/alluxio/heartbeat/ScheduledTimer.java

```diff
@@ -69,7 +72,8 @@ protected void schedule() {
     try (LockResource r = new LockResource(mLock)) {
       Preconditions.checkState(!mScheduled, "Called schedule twice without waiting for any ticks");
       mScheduled = true;
-      mCondition.signal();
+      mTickCondition.signal();
+      HeartbeatScheduler.removeTimer(this);
```

Comment:
There aren't any expected exceptions that `signal` could throw, but if it throws an unexpected exception it will propagate and fail the test.